[CodeStyle][task 7] enable Ruff UP031 rule in python/paddle/base

Kaedeharai opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Others
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
Others
### Description
<!-- Describe what you’ve done -->
### Related links
 
- #57367
@gouzil

Kaedeharai committed:
delete UP031

gouzil commented:
需要再按照 UP032 修一下
```bash
ruff check .
python/paddle/base/core.py:50:13: UP032 [*] Use f-string instead of `format` call
python/paddle/base/core.py:199:13: UP032 [*] Use f-string instead of `format` call
python/paddle/base/executor.py:488:16: UP032 [*] Use f-string instead of `format` call
python/paddle/base/executor.py:511:20: UP032 [*] Use f-string instead of `format` call
python/paddle/base/executor.py:2793:20: UP032 [*] Use f-string instead of `format` call
python/paddle/base/framework.py:1879:28: UP032 [*] Use f-string instead of `format` call
python/paddle/base/framework.py:3045:33: UP032 [*] Use f-string instead of `format` call
python/paddle/base/framework.py:3053:33: UP032 [*] Use f-string instead of `format` call
python/paddle/base/framework.py:3099:29: UP032 [*] Use f-string instead of `format` call
python/paddle/base/framework.py:3117:37: UP032 [*] Use f-string instead of `format` call
python/paddle/base/framework.py:7292:28: UP032 [*] Use f-string instead of `format` call
python/paddle/base/trainer_desc.py:239:17: UP032 [*] Use f-string instead of `format` call
python/paddle/base/trainer_desc.py:255:17: UP032 [*] Use f-string instead of `format` call
python/paddle/base/trainer_desc.py:273:17: UP032 [*] Use f-string instead of `format` call
```
用`ruff check . --fix`就行

Kaedeharai committed:
[CodeStyle][task 7] enable Ruff UP031 rule in python/paddle/base

Kaedeharai committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into second

gouzil commented:
其他的风格检测工具没过，需要按照[文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/code_contributing_path_cn.html#ergongxianliucheng)配置一下`pre-commit`

Kaedeharai committed:
modified:   python/paddle/base/backward.py
	modified:   python/paddle/base/data_feeder.py
	modified:   python/paddle/base/dygraph/base.py
	modified:   python/paddle/base/executor.py
	modified:   python/paddle/base/framework.py
	modified:   python/paddle/base/layers/layer_function_generator.py

Kaedeharai committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/Paddle into second

gouzil commented:
LGTM，@SigureMo

SigureMo reviewed (approved):
LGTMeow 🐾
